### DISCUSSION TITLE
[Bug] When starting Trino with a pulsar plugin using docker, an error of incompatible JDK version appears.

### DISCUSSION THREAD

**Discussion by crossoverJie** [apache/pulsar#20941] [Q&A] [answered]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

Pulsar: 3.0.0
Trino: 368

#### Download plugin:
```shell
wget https://archive.apache.org/dist/pulsar/pulsar-3.0.0/apache-pulsar-3.0.0-bin.tar.gz
```

#### Copy plugin into image:
```
FROM trinodb/trino:368
COPY pulsar-presto-connector /usr/lib/trino/plugin/pulsar-presto-connector
```

Start container:
```
docker build . -t trinodb/trino:368-pulsar-3.0.0
docker run --name trino -d -p 8080:8080 trinodb/trino:368-pulsar-3.0.0
```

### What did you expect to see?

The container started successfully.

### What did you see instead?

![WeChatWorkScreenshot_2eee8398-ed20-42f3-8eab-7dac78884322](https://github.com/apache/pulsar/assets/15684156/d985893c-41bd-4830-acac-226caf2762de)

### Anything else?

When I used `Trino:latest` and `Pulsar:3.0.0`, got an error of `NoclassDefFoundError`:
![WeChatWorkScreenshot_e60eae53-57b1-4cb2-a40b-41d9165315a9](https://github.com/apache/pulsar/assets/15684156/28001550-502a-426c-8e28-930e51806573)

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by tisonkun** [comment, marked as answer]
Here you come into two issues:

1. Trino 368 is compiled with JDK 11 while our plugin is compiled with JDK 17. You can try to compile the plugin with JDK 11 but it's not promised to work.
2. Trino lastest makes a lot of breaking changes from the one (368) used in our plugin. You can take a look at https://github.com/apache/pulsar/pull/16494#issue-1299646843 and https://github.com/apache/pulsar/pull/20016 for some background.

Basically, please run the Trino worker from the one bundled by Pulsar distro.

Both of them are backed in the root that Pulsar SQL is not under primary maintenance (as well as that Trino pays less attention of compatibility). I suppose anyone who can cover its maintenance a qualified committer.

**Reply by crossoverJie** [reply]
Thank you for your answer.

1. I previously tried to use the JDK11 compiler plugin, but it failed.
2. You are right, so what we are doing now is compiling the plugin source code and manually building a Trino image to solve it.

Adding a Dockerfile under the `pulsar-sql` module and building your own image from the source code, is that a better solution?
